build.rs

```diff
@@ -433,7 +435,9 @@ fn build_c_code(
 
     generate_prefix_symbols_asm_headers(out_dir, ring_core_prefix).unwrap();
 
-    let (asm_srcs, obj_srcs) = if let Some(asm_target) = asm_target {
+    let (asm_srcs, obj_srcs) = if let Some(asm_target) = find_asm_target(target) {
+        println!("cargo:rustc-cfg=have_perlasm");
```

Comment:
I suggest we name it just `perlasm`.

I am worried about the risk for users of non-Cargo build systems. I suggest when we do not set `perlasm`, we set `no_perlasm`. Then we should add a check like this to lib.rs to ensure that the such users have addressed this:
```
const PERLASM_CONFIGURED: () = assert!((cfg!(perlasm) && !cfg!(no_perlasm)) || (!cfg!(perlasm) && cfg!(no_perlasm)));
```

Reply:
Good point. I've implemented this.